src/instructlab/dolomite/hf_models/models/gpt_dolomite/config.py

```diff
@@ -1,9 +1,128 @@
 # ----------------------------------------------------------------
 # Extracted from https://github.com/ibm-granite/dolomite-engine
 # ----------------------------------------------------------------
+# Third Party
+from transformers import PretrainedConfig
+
 # Local
-from ...config import CommonConfig
+from ...enums import AttentionHeadType, PositionEmbeddingType
 
 
-class GPTDolomiteConfig(CommonConfig):
+class GPTDolomiteConfig(PretrainedConfig):
     model_type = "gpt_dolomite"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {
+        "hidden_size": "n_embd",
+        "max_position_embeddings": "n_positions",
+        "num_attention_heads": "n_head",
+        "num_hidden_layers": "n_layer",
+    }
+
+    # NOTE: initializer range is kept for backward compatiblity
+    #       but it is not used anymore
+
+    def __init__(
+        self,
+        vocab_size: int = 50257,
+        n_positions: int = 1024,
+        n_embd: int = 768,
+        n_layer: int = 12,
+        n_head: int = 12,
+        num_key_value_heads: int = None,
+        n_inner: int = None,
+        activation_function: str = "gelu_pytorch_tanh",
+        attention_head_type: str = "mqa",
+        resid_pdrop: float = 0.1,
+        embd_pdrop: float = 0.1,
+        attn_pdrop: float = 0.1,
+        normalization_function: str = "layernorm",
+        layer_norm_epsilon: float = 1e-5,
+        initializer_range: float = 0.02,
+        scale_attn_weights: bool = True,
+        attention_multiplier: float = None,
+        use_cache: bool = True,
+        bos_token_id: int = 50256,
+        eos_token_id: int = 50256,
+        pad_token_id: int = 50256,
+        attention_softmax_in_fp32: bool = True,
+        scale_attention_softmax_in_fp32: bool = True,
+        add_bias: bool = True,
+        position_embedding_type: str = "learned_absolute",
+        rope_theta: int = 10000,
+        rope_scaling: dict = None,
```

Comment:
we can drop rope_scaling

Reply:
im just worried to drop if, as we have checkpoints trained on https://github.com/ibm-granite/dolomite-engine/tree/main/dolomite_engine, and we may lose backward compatibility if i delete this key

Reply:
are you ok if I just keep it first, but will put a comment to eventually drop it at some point?

Reply:
added the comment in https://github.com/instructlab/GPTDolomite/pull/3/commits/8afe81d60758e41ab00e1a98d4f86d3282e8b002